src/controllers/AuthController.ts

```diff
@@ -106,264 +104,122 @@ class AuthController {
 
 			const data = await User.create(inputData)
 
-			return Response.successResponse({
+			return response.successResponse({
 				message: "User created successfully",
 				data
 			})
-		} catch (error: any) {
+		} catch (error) {
 			next(error)
 		}
 	}
 
 	public async sendOtp(req: Request, res: Response, next: NextFunction) {
-		// try {
-		// 	const Response = new ApiResponse(res)
-		// 	const {email}: SendOtpPayload = req.body
-		// //check if user exist
-		// const userExists: UserDetails | null =
-		// 	await User.findOne({
-		// 		email,
-		// 		isDeleted: false,
-		// 		isActive: true
-		// 	})
-		// if (!userExists)
-		// {
-		// 	return Response.errorResponse({
-		// 		...errorData.NOT_FOUND,
-		// 		message: "User not found"
-		// 	})
-		// }
-		// 	// generate otp
-		// 	const otp: number = await generateOtp()
-		// 	// create encryption
-		// 	const encryptedOtp: string = jwt.sign(
-		// 		{
-		// 			otp
-		// 		},
-		// 		process.env.JWT_SECRET_KEY as string
-		// 	)
-		// 	// save encrypted otp
-		// 	const verificationDataExists: VerificationDetails[] =
-		// 		await this.verificationCommonModel.list({value: userName})
-		// 	if (!verificationDataExists || !verificationDataExists?.length) {
-		// 		await this.verificationCommonModel.bulkCreate([
-		// 			{
-		// 				verificationType: authCredential.logInWith,
-		// 				value: userName,
-		// 				otp: encryptedOtp,
-		// 				isVerified: false,
-		// 				verificationFor: VerificationFor.AUTH
-		// 			}
-		// 		])
-		// 	} else {
-		// 		await this.verificationCommonModel.update(
-		// 			{otp: encryptedOtp},
-		// 			verificationDataExists[0]?.verificationId
-		// 		)
-		// 	}
-		// 	// get first name of the user
-		// 	const [userDetails]: UserShortDetails[] =
-		// 		await this.commonModel.list({
-		// 			userId: authCredential.userId
-		// 		})
-		// 	if (!userDetails) {
-		// 		throw new UnauthorizedException("User not found")
-		// 	}
-		// 	if (authCredential.logInWith === LogInWith.MOBILE) {
-		// 		// send sms
-		// 		// await sendSMS(authCredential[0].mobile, otp);
-		// 	} else if (authCredential.logInWith === LogInWith.EMAIL) {
-		// 		// send otp to email
-		// 		await sendOtpToEmail(
-		// 			authCredential.userName,
-		// 			otp,
-		// 			userDetails.firstName
-		// 		)
-		// 	}
-		// 	return res.json({
-		// 		success: true,
-		// 		message: `OTP sent successfully`
-		// 	})
-		// } catch (error) {
-		// 	res.status(400).json({
-		// 		status: 400,
-		// 		message: error?.toString(),
-		// 		code: "unexpected_error"
-		// 	})
-		// 	return
-		// }
+		try {
+			const response = new ApiResponse(res)
+			const {email}: SendOtpPayload = req.body
+			//check if user exist
+			const userExists: UserDetails | null = await User.findOne({
+				email,
```

Comment:
Please save the otp in secrectCode field in user model.
Do not forget to create hash before save it.
for creating hash first use both otp and timestamp.
Don't forget to add env time limit to the current timestamp.